### PR TITLE
add auth-type oauth2

### PR DESCRIPTION
**Pull request by chenwx36**

### CODE REVIEW

**Comment by adolli** [comment]
**changes in a summary.**
- add new auth type oauth2
- unify some filesystem operation and use request path to represent the path in filesystem

File: httpstaticserver.go [hunk 1]
```diff
@@ -83,28 +83,53 @@ func NewHTTPStaticServer(root string) *HTTPStaticServer {
...
 	return s
 }
 
 func (s *HTTPStaticServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s.m.ServeHTTP(w, r)
 }
 
+func (s *HTTPStaticServer) hGet(w http.ResponseWriter, r *http.Request) {
+	if r.FormValue("op") == "info" {
+		 fmt.Println("hGet op=info")
```
Comment (codeskyblue) [inline httpstaticserver.go]: remove debug message

File: httpstaticserver.go [hunk 2]
```diff
@@ -83,28 +83,53 @@ func NewHTTPStaticServer(root string) *HTTPStaticServer {
...
+		 s.hInfo(w, r)
+	}else {
+		fmt.Println("hGet op not equal info")
```
Comment (codeskyblue) [inline httpstaticserver.go]: remove debug message

File: httpstaticserver.go [hunk 3]
```diff
@@ -83,28 +83,53 @@ func NewHTTPStaticServer(root string) *HTTPStaticServer {
...
+func (s *HTTPStaticServer) hPOST(w http.ResponseWriter, r *http.Request) {
+	if r.FormValue("op") == "mkdir" {
+		s.hMkdir(w, r)
+	}else {
+		s.hUpload(w, r)
+	}
+	return
+}
+
+func (s *HTTPStaticServer) hDELETE(w http.ResponseWriter, r *http.Request) {
```
Comment (codeskyblue) [inline httpstaticserver.go]: useless code

File: httpstaticserver.go [hunk 4]
```diff
@@ -83,28 +83,53 @@ func NewHTTPStaticServer(root string) *HTTPStaticServer {
...
+	}else {
+		fmt.Println("hGet op not equal info")
+		s.hIndex(w, r)
+	}
+	return
+}
+
+func (s *HTTPStaticServer) hPOST(w http.ResponseWriter, r *http.Request) {
+	if r.FormValue("op") == "mkdir" {
+		s.hMkdir(w, r)
```
Comment (codeskyblue) [inline httpstaticserver.go]: need auth check

**Pull request closed without merging**